File: 教师端/teacher-java/service/src/main/java/com/system/service/profile/entity/TeacherProfile.java
package com.system.service.profile.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;


@Data
@TableName("teacher_profile")
public class TeacherProfile implements Serializable {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long teacherId;

    private String name;

    private String avatar;

    private String email;

    private String phone;

    private Integer gender;

    private String specialty;

    private String bio;

    private Integer active;

    private String department;

    private Integer teachingYears;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: 教师端/my-vue-app/src/types/TeacherProfile.ts
/**
 * 教师档案接口类型定义
 */
export interface TeacherProfile {
    id?: number;
    teacherId?: number;
    name: string;
    avatar?: string;
    email?: string;
    phone?: string;
    gender?: number; // 0-保密，1-男，2-女
    specialty?: string;
    bio?: string;
    active?: number; // 0-禁用，1-启用
    department?: string;
    teachingYears?: number;
    createdAt?: string;
    updatedAt?: string;
}
